service/src/main/java/greencity/service/CommentServiceImpl.java

```diff
@@ -414,6 +414,27 @@ public int countAllActiveReplies(Long parentCommentId) {
         return commentRepo.countByParentCommentIdAndStatusNot(parentCommentId, CommentStatus.DELETED);
     }
 
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public PageableDto<CommentDto> getAllComments(Pageable pageable, ArticleType articleType, Long articleId,
+        UserVO user, List<CommentStatus> statuses) {
+        if (articleType == ArticleType.HABIT) {
+            Optional<Habit> habit = habitRepo.findById(articleId);
+            if (habit.isEmpty()) {
+                throw new NotFoundException(HABIT_NOT_FOUND_BY_ID + articleId);
+            }
+        }
+
+        Page<Comment> pages = commentRepo.findAllByArticleTypeAndArticleIdAndStatusInOrderByCreatedDateDesc(pageable,
+            articleType, articleId, statuses);
+
+        pages = setCurrentUserLiked(pages, user);
+
+        return convertPagesToCommentDtos(pages, user);
+    }
+
```

Comment:
```java
    /**
     * {@inheritDoc}
     */
    @Override
    public PageableDto<CommentDto> getAllActiveComments(Pageable pageable, UserVO userVO, Long articleId,
        ArticleType articleType) {
        if (articleType == ArticleType.HABIT) {
            Optional<Habit> habit = habitRepo.findById(articleId);
            if (habit.isEmpty()) {
                throw new NotFoundException(HABIT_NOT_FOUND_BY_ID + articleId);
            }
        }

        Page<Comment> pages =
            commentRepo.findAllByParentCommentIdIsNullAndArticleIdAndArticleTypeAndStatusNotOrderByCreatedDateDesc(
                pageable, articleId, articleType, CommentStatus.DELETED);

        pages = setCurrentUserLiked(pages, userVO);

        return convertPagesToCommentDtos(pages, userVO);
    }
    
        /**
     * {@inheritDoc}
     */
    @Override
    public PageableDto<CommentDto> getAllActiveReplies(Pageable pageable, Long parentCommentId, UserVO userVO) {
        Page<Comment> pages =
            commentRepo.findAllByParentCommentIdAndStatusNotOrderByCreatedDateDesc(pageable, parentCommentId,
                CommentStatus.DELETED);

        pages = setCurrentUserLiked(pages, userVO);

        return convertPagesToCommentDtos(pages, userVO);
    }

    ```
    
do we need this methods which are returting seperately comments and seperately replies to them if all Comments [comments+replies] would be returned together?

Reply:
I think we don’t need these two separate methods for fetching comments and replies. It would be more efficient to return all comments (including replies) together in one call. This simplifies the code, reduces the number of methods to maintain, and avoids redundancy. In most use cases, we need both comments and replies displayed together, so it makes sense to handle them in a unified way rather than separately.

I don't understand the thought process of the previous developer in developing two methods instead of one. Maybe it's a good idea, but why? So, I’ve decided to leave them for now, just in case there’s a specific reason that we haven’t considered yet.